### PR TITLE
Add package creation for MacOS

### PR DESCRIPTION
**Pull request by terjekv**

### CODE REVIEW

**Comment by kniec** [comment]
AWS has mac1.metal instances now... :)

File: .github/workflows/build-client-packages.yml [hunk 1]
```diff
@@ -54,6 +57,13 @@ jobs:
           fpm_args: "etc"
           fpm_opts: "--debug -n cvmfs-config-eessi -v ${{ steps.get_version.outputs.version }} -t deb -a all -s dir -C ./package --description 'CVMFS config repository package for EESSI.'"
 
+      - name: Build MacOS package
+        id: build-osxpkg
+        uses: bpicode/github-action-fpm@master
+        with:
+          fpm_args: "etc"
+          fpm_opts: "--debug -n cvmfs-config-eessi -v ${{ steps.get_version.outputs.version }} -t osxpkg -a all -s dir -C ./package --description 'CVMFS config repository package for EESSI.'"
```
Comment (boegel) [inline .github/workflows/build-client-packages.yml:65]: We could consider using Travis for this, they have Mac CI support...

Travis isn't very interesting these days though, only a very limited amount of credits available (although we don't need much in this case)

Comment (bedroge) [reply .github/workflows/build-client-packages.yml:65]: Or a separate Github Action? They seem to support it as well:
https://docs.github.com/en/free-pro-team@latest/actions/reference/specifications-for-github-hosted-runners#supported-runners-and-hardware-resources

Comment (boegel) [reply .github/workflows/build-client-packages.yml:65]: Ah, that's new...

Doesn't need to be a separate action, you can just run the same one on different platforms (linux/macos), and use conditionals to opt-in/out of steps, see https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepsif

Comment (bedroge) [reply .github/workflows/build-client-packages.yml:65]: Seems to work:
https://github.com/bedroge/filesystem-layer/runs/1497929661?check_suite_focus=true

Comment (terjekv) [reply .github/workflows/build-client-packages.yml:65]: Neat! Although this warning... https://github.com/bedroge/filesystem-layer/runs/1497929661?check_suite_focus=true#step:5:4 It won't ever go away, will it?

Comment (boegel) [reply .github/workflows/build-client-packages.yml:65]: You can use `pip3 install ansible` instead.

Comment (bedroge) [reply .github/workflows/build-client-packages.yml:65]: Yep, it's gone now.

File: .github/workflows/build-client-packages.yml [hunk 2]
```diff
@@ -4,20 +4,19 @@ on:
...
     tags:
       - 'v*'
   pull_request:
     branches:
       - master
-      - build_packages
 
 jobs:
 
   build-packages:
```
Comment (boegel) [inline .github/workflows/build-client-packages.yml:15]: rename to `build-linux-packages`?

File: .github/workflows/build-client-packages.yml [hunk 3]
```diff
@@ -75,8 +77,57 @@ jobs:
...
+
+      - name: Upload macOS package as artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: macOS package
+          path: ${{ steps.find_filenames.outputs.pkgfile }}
+
   release:
-    needs: build-packages
+    needs: [build-packages, build-osx-package]
```
Comment (boegel) [inline .github/workflows/build-client-packages.yml:130]: also rename `build-packages` to `build-linux-packages` here

**Review by boegel — CHANGES_REQUESTED** [review]

**Review by boegel — APPROVED** [review]

**Comment by boegel** [comment]
@bedroge Failing tests look unrelated to the changes in this PR?

**Comment by bedroge** [comment]
@boegel The Centos 8 failures/timeouts have been solved in #57.

But I don't understand the issue with build-linux-packages: fpm gave some weird error
```
/usr/local/bundle/gems/fpm-1.11.0/lib/fpm/package/deb.rb:495:in `block in output': uninitialized constant FPM::Package::Deb::Zlib (NameError)
171	from <internal:kernel>:90:in `tap'
172	from /usr/local/bundle/gems/fpm-1.11.0/lib/fpm/package/deb.rb:494:in `output'
173	from /usr/local/bundle/gems/fpm-1.11.0/lib/fpm/command.rb:487:in `execute'
174	from /usr/local/bundle/gems/clamp-1.0.1/lib/clamp/command.rb:68:in `run'
175	from /usr/local/bundle/gems/fpm-1.11.0/lib/fpm/command.rb:574:in `run'
176	from /usr/local/bundle/gems/clamp-1.0.1/lib/clamp/command.rb:133:in `run'
177	from /usr/local/bundle/gems/fpm-1.11.0/bin/fpm:7:in `<top (required)>'
178	from /usr/local/bundle/bin/fpm:23:in `load'
179	from /usr/local/bundle/bin/fpm:23:in `<main>'
```
This worked fine with the last commits... and it doesn't look like the fpm action has been changed in the meantime (fpm version is still the same too)....

**Comment by bedroge** [comment]
https://github.com/jordansissel/fpm/issues/1739
https://github.com/jordansissel/fpm/blob/v1.11.0/lib/fpm/package/deb.rb#L495

I have no clue how this could have worked before...

**Comment by bedroge** [comment]
The CI for building the packages has been fixed, see https://github.com/EESSI/filesystem-layer/issues/64.
Rerunning the tests now...

A fix for the failing Centos 8 tests is included in this PR:
https://github.com/EESSI/filesystem-layer/pull/57

**Comment by bedroge** [comment]
Hmm, I think this part was already included in #48, so I guess we can close this PR.

**Comment by bedroge** [comment]
See for instance this action (triggered by PR #57) for a macOS package:
https://github.com/EESSI/filesystem-layer/actions/runs/408367869
We can make a new release soon, so that we have a versioned package for macOS.

Closing this PR.

**Pull request closed without merging**